bundles/org.eclipse.equinox.p2.ui.discovery/src/org/eclipse/equinox/internal/p2/ui/discovery/operations/DiscoveryInstallOperation.java

```diff
@@ -93,7 +93,9 @@ public void run(IProgressMonitor progressMonitor) throws InvocationTargetExcepti
 				monitor.done();
 			}
 		} catch (OperationCanceledException e) {
-			throw new InterruptedException();
+			InterruptedException interruptedException = new InterruptedException(e.getLocalizedMessage());
```

Comment:
Can't the catched exception directly be passed to the constructor of `InterruptedException`?